packages/core/test/financial-templates/perpetual-multiparty/PerpetualPositionManager.js

```diff
@@ -1016,6 +1029,242 @@ contract("PerpetualPositionManager", function(accounts) {
+    await collateral.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    await collateral.approve(positionManager.address, toWei("100000"), { from: other });
+    await tokenCurrency.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    await tokenCurrency.approve(positionManager.address, toWei("100000"), { from: other });
+    await positionManager.create({ rawValue: toWei("150") }, { rawValue: toWei("100") }, { from: sponsor });
+
+    assert.equal((await positionManager.cumulativeFundingRateMultiplier()).toString(), toWei("1.05"));
+
+    // Set the funding rate to a negative funding rate of 0.98 in the store and apply it for 5 seconds. New funding rate
+    // should be 1.05 * (1 - (1 - 0.98) * 5) = 0.945
+    await mockFundingRateStore.setFundingRate(priceFeedIdentifier, await timer.getCurrentTime(), {
+      rawValue: toWei("0.98")
+    });
+    await timer.setCurrentTime((await timer.getCurrentTime()).add(toBN(5)).toString()); // Advance the time by 5 seconds
+    await positionManager.requestWithdrawal({ rawValue: toWei("10") }, { from: sponsor }); // Requesting withdraw should also update funding multipler
+    assert.equal((await positionManager.cumulativeFundingRateMultiplier()).toString(), toWei("0.945"));
+
+    // Setting the funding rate to zero (no payments made, synth trading at parity) should no change the cumulativeFundingRateMultiplier.
+    await mockFundingRateStore.setFundingRate(priceFeedIdentifier, await timer.getCurrentTime(), {
+      rawValue: toWei("1")
+    });
+    await timer.setCurrentTime((await timer.getCurrentTime()).add(toBN(withdrawalLiveness)).toString()); // Advance the time by the withdrawal liveness
+    await positionManager.withdrawPassedRequest({ from: sponsor }); // call another function on the contract.
+    assert.equal((await positionManager.cumulativeFundingRateMultiplier()).toString(), toWei("0.945"));
+
+    // Check that the remaining functions update the funding rate accordingly.
+    // Have already checked: a) create b) requestWithdrawal and c) withdrawPassedRequest
+    await mockFundingRateStore.setFundingRate(priceFeedIdentifier, await timer.getCurrentTime(), {
+      rawValue: toWei("1.01")
+    });
+
+    // requestTransferPosition
+    await timer.setCurrentTime((await timer.getCurrentTime()).add(toBN(1)).toString());
+    await positionManager.requestTransferPosition({ from: sponsor });
+    assert.equal((await positionManager.cumulativeFundingRateMultiplier()).toString(), toWei("0.95445")); // 0.945 * (1 + (1.01 - 1) * 1) = 0.95445
+
+    // requestTransferPosition. For this we will need to advance time by the withdrawlLiveness to pass the request withdraw.
+    // Set the funding rate to something more reasonable like 0.001% per second as 1.0001 as advancing over a long duration.
+    await mockFundingRateStore.setFundingRate(priceFeedIdentifier, await timer.getCurrentTime(), {
```

Comment:
WDYT about defining a helper function `setRateAndAdvanceTimer(rate, time)` that handles the `setFundingRate()` and `setCurrentTime()` repeated logic?

Reply:
we could refactor this but `setFundingRate` is only called at the top before the bulk of the tests and many functions require a slightly different setup such as withdraw requiring a new setup. IMHO it will be a bit messy to try to generalize this as 7 out of the 15 functions called in this test would require exception handling. can still refactor if you'd like.

Reply:
Got it